src/Orchard.Environment.Extensions/Compilers/CSharpExtensionCompiler.cs

```diff
@@ -1,65 +1,128 @@
+                return _compiledLibraries[context.RootProject.Identity.Name];
+            }
+
+            // Get compilation options
+            var compilationOptions = context.ResolveCompilationOptions(config);
+            var compileInclude = context.ResolveCompilationOptions(config).CompileInclude;
+
+            var projectSourceFiles = new List<string>();
+
+            // Get project source files
+            if (compilationOptions.CompileInclude == null)
+            {
+                projectSourceFiles.AddRange(context.ProjectFile.Files.SourceFiles);
+            }
+            else {
+                var includeFiles = IncludeFilesResolver.GetIncludeFiles(compilationOptions.CompileInclude, "/", diagnostics: null);
+                projectSourceFiles.AddRange(includeFiles.Select(f => f.SourcePath));
+            }
+
+            // Check if precompiled
+            if (!projectSourceFiles.Any())
+            {
+                return _compiledLibraries[context.RootProject.Identity.Name] = true;
+            }
+
+           // Set up Output Paths
+            var outputPaths = context.GetOutputPaths(config);
             var outputPath = outputPaths.CompilationOutputPath;
             var intermediateOutputPath = outputPaths.IntermediateOutputDirectoryPath;
 
             Directory.CreateDirectory(outputPath);
             Directory.CreateDirectory(intermediateOutputPath);
 
             // Create the library exporter
-            var exporter = context.CreateExporter(config, buildBasePath);
+            var exporter = context.CreateExporter(config);
 
             // Gather exports for the project
             var dependencies = exporter.GetDependencies().ToList();
 
             var diagnostics = new List<DiagnosticMessage>();
-            var missingFrameworkDiagnostics = new List<DiagnosticMessage>();
 
             // Collect dependency diagnostics
             foreach (var diag in context.LibraryManager.GetAllDiagnostics())
             {
-                if (diag.ErrorCode == ErrorCodes.DOTNET1011 ||
-                    diag.ErrorCode == ErrorCodes.DOTNET1012)
-                {
-                    missingFrameworkDiagnostics.Add(diag);
-                }
+                // Some library paths may not be resolved by the project model (e.g in production)
+                // So, here we don't grab any diagnostics and we will search in the probing folder
 
-                diagnostics.Add(diag);
+                //Diagnostics.Add(diag.FormattedMessage);
+                //diagnostics.Add(diag);
             }
 
-            if(diagnostics.Any(d => d.Severity == DiagnosticMessageSeverity.Error))
+            if (diagnostics.Any(d => d.Severity == DiagnosticMessageSeverity.Error))
             {
                 // We got an unresolved dependency or missing framework. Don't continue the compilation.
-                return false;
+                return _compiledLibraries[context.RootProject.Identity.Name] = false;
```

Comment:
Here, we skip the extension / dependency project compilation. Then, right now, it depends on an already compiled assembly in a binary / probing folder ...